### PR TITLE
Configuration defaults for a better out of the box experience

### PR DESCRIPTION
**Pull request by tinchou**
**TL;DR;** we can now do this:

* Launch mainnet: `java -cp path/to/node.jar co.rsk.Start`
* Launch testnet: `java -cp path/to/node.jar -Dblockchain.config.name=testnet co.rsk.Start`
* Launch regtest: `java -cp path/to/node.jar -Dblockchain.config.name=regtest co.rsk.Start`

### Details

* Adds a `reference.conf` file, with shared configuration for all networks
* Adds `config/[network].conf`, one per network, with defaults that work out of the box
* The node joins mainnet by default
* A different network can be chosen with just a system property (e.g. `java -cp path/to/jar -Dblockchain.config.name=testnet`, to be improved in a future PR)
* By default, the database dir is now `$HOME/.rsk/[network]/database`
* All configuration options can be overrode:
  * Providing a path to a config file with overrides (`-Drsk.conf.file path/to/file.conf`)
  * Through explicit system properties (`-Drpc.enabled=false`), which in turn take precedence over any other configuration

Improvements in the base configuration files (secure defaults, sane defaults, better documentation) are also scheduled for a future PR (@herrerameri you'll have to help me with that!)

### CODE REVIEW

File: rskj-core/src/main/resources/config/testnet.conf [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+            "bootstrap02.testnet.rsk.co:50505",
+            "bootstrap03.testnet.rsk.co:50505",
+            "bootstrap04.testnet.rsk.co:50505",
+            "bootstrap05.testnet.rsk.co:50505",
+            "bootstrap06.testnet.rsk.co:50505"
+        ]
+    }
+
+    # Port for server to listen for incoming connections
+    port = 5050
```
Comment (bcodesido) [inline rskj-core/src/main/resources/config/testnet.conf]: Testnet default port is 50505

File: rskj-core/src/main/resources/reference.conf [hunk 2]
```diff
@@ -0,0 +1,237 @@
...
+
+    # connection timeout for trying to connect to a peer [seconds]
+    connection.timeout = 2
+
+    # the parameter specifies how much time we will wait for a message to come before closing the channel
+    channel.read.timeout = 30
+
+    # Private key of the peer
+    # nodeId = <NODE_ID>
+    # privateKey = <PRIVATE_KEY>
```
Comment (bcodesido) [inline rskj-core/src/main/resources/reference.conf]: shouldn't be uncommented?

Comment (tinchou) [reply rskj-core/src/main/resources/reference.conf]: Not really, we are using the existing feature of auto-generating a `nodeId/PK`

File: rskj-core/src/main/resources/config/testnet.conf (hunk 1 above)
Comment (tinchou) [reply rskj-core/src/main/resources/config/testnet.conf]: Done

**Comment by rskops** [comment]
SonarQube analysis reported 69 issues
* ![CRITICAL][CRITICAL] 1 critical
* ![MAJOR][MAJOR] 39 major
* ![MINOR][MINOR] 8 minor
* ![INFO][INFO] 21 info

#### Top 10 issues

1. ![CRITICAL][CRITICAL] [SystemProperties.java#L164](https://github.com/rsksmart/rskj/blob/94ba6dd5b601f0da4857b1ca5e9d0edf99005ca4/rskj-core/src/main/java/org/ethereum/config/SystemProperties.java#L164): Make the enclosing method "static" or remove this set. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS2696)
1. ![MAJOR][MAJOR] [RskSystemProperties.java#L298](https://github.com/rsksmart/rskj/blob/94ba6dd5b601f0da4857b1ca5e9d0edf99005ca4/rskj-core/src/main/java/co/rsk/config/RskSystemProperties.java#L298): Method co.rsk.config.RskSystemProperties.getRpcModules() does not presize the allocation of a collection [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3APSC_PRESIZE_COLLECTIONS)
1. ![MAJOR][MAJOR] [RskSystemProperties.java#L332](https://github.com/rsksmart/rskj/blob/94ba6dd5b601f0da4857b1ca5e9d0edf99005ca4/rskj-core/src/main/java/co/rsk/config/RskSystemProperties.java#L332): java/nio/file/Paths.get(Ljava/lang/String;[Ljava/lang/String;)Ljava/nio/file/Path; reads a file whose location might be specified by user input [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=findsecbugs%3APATH_TRAVERSAL_IN)
1. ![MAJOR][MAJOR] [RskSystemProperties.java#L333](https://github.com/rsksmart/rskj/blob/94ba6dd5b601f0da4857b1ca5e9d0edf99005ca4/rskj-core/src/main/java/co/rsk/config/RskSystemProperties.java#L333): java/nio/file/Paths.get(Ljava/lang/String;[Ljava/lang/String;)Ljava/nio/file/Path; reads a file whose location might be specified by user input [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=findsecbugs%3APATH_TRAVERSAL_IN)
1. ![MAJOR][MAJOR] [RskSystemProperties.java#L335](https://github.com/rsksmart/rskj/blob/94ba6dd5b601f0da4857b1ca5e9d0edf99005ca4/rskj-core/src/main/java/co/rsk/config/RskSystemProperties.java#L335): java/nio/file/Paths.get(Ljava/lang/String;[Ljava/lang/String;)Ljava/nio/file/Path; reads a file whose location might be specified by user input [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=findsecbugs%3APATH_TRAVERSAL_IN)
1. ![MAJOR][MAJOR] [RskSystemProperties.java#L343](https://github.com/rsksmart/rskj/blob/94ba6dd5b601f0da4857b1ca5e9d0edf99005ca4/rskj-core/src/main/java/co/rsk/config/RskSystemProperties.java#L343): java/io/FileOutputStream.<init>(Ljava/lang/String;)V writes to a file whose location might be specified by user input [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=findsecbugs%3APATH_TRAVERSAL_OUT)
1. ![MAJOR][MAJOR] [DefaultConfig.java#L75](https://github.com/rsksmart/rskj/blob/94ba6dd5b601f0da4857b1ca5e9d0edf99005ca4/rskj-core/src/main/java/org/ethereum/config/DefaultConfig.java#L75): java/io/File.<init>(Ljava/lang/String;)V reads a file whose location might be specified by user input [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=findsecbugs%3APATH_TRAVERSAL_IN)
1. ![MAJOR][MAJOR] [SystemProperties.java#L87](https://github.com/rsksmart/rskj/blob/94ba6dd5b601f0da4857b1ca5e9d0edf99005ca4/rskj-core/src/main/java/org/ethereum/config/SystemProperties.java#L87): Method org.ethereum.config.SystemProperties.<static initializer for SystemProperties>() needlessly boxes a boolean constant [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3ANAB_NEEDLESS_BOOLEAN_CONSTANT_CONVERSION)
1. ![MAJOR][MAJOR] [SystemProperties.java#L119](https://github.com/rsksmart/rskj/blob/94ba6dd5b601f0da4857b1ca5e9d0edf99005ca4/rskj-core/src/main/java/org/ethereum/config/SystemProperties.java#L119): Incorrect lazy initialization and update of static field org.ethereum.config.SystemProperties.configFromFiles in new org.ethereum.config.SystemProperties(ConfigLoader) [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=findbugs%3ALI_LAZY_INIT_UPDATE_STATIC)
1. ![MAJOR][MAJOR] [SystemProperties.java#L121](https://github.com/rsksmart/rskj/blob/94ba6dd5b601f0da4857b1ca5e9d0edf99005ca4/rskj-core/src/main/java/org/ethereum/config/SystemProperties.java#L121): Method new org.ethereum.config.SystemProperties(ConfigLoader) passes a concatenated string to SLF4J's format string [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3ALO_APPENDED_STRING_IN_FORMAT_STRING)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by rskops** [comment]
SonarQube analysis reported 69 issues
* ![CRITICAL][CRITICAL] 1 critical
* ![MAJOR][MAJOR] 39 major
* ![MINOR][MINOR] 8 minor
* ![INFO][INFO] 21 info

#### Top 10 issues

1. ![CRITICAL][CRITICAL] [SystemProperties.java#L164](https://github.com/rsksmart/rskj/blob/d91111a0570e100175143f2b9b63c0669db7ce03/rskj-core/src/main/java/org/ethereum/config/SystemProperties.java#L164): Make the enclosing method "static" or remove this set. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS2696)
1. ![MAJOR][MAJOR] [RskSystemProperties.java#L298](https://github.com/rsksmart/rskj/blob/d91111a0570e100175143f2b9b63c0669db7ce03/rskj-core/src/main/java/co/rsk/config/RskSystemProperties.java#L298): Method co.rsk.config.RskSystemProperties.getRpcModules() does not presize the allocation of a collection [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3APSC_PRESIZE_COLLECTIONS)
1. ![MAJOR][MAJOR] [RskSystemProperties.java#L332](https://github.com/rsksmart/rskj/blob/d91111a0570e100175143f2b9b63c0669db7ce03/rskj-core/src/main/java/co/rsk/config/RskSystemProperties.java#L332): java/nio/file/Paths.get(Ljava/lang/String;[Ljava/lang/String;)Ljava/nio/file/Path; reads a file whose location might be specified by user input [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=findsecbugs%3APATH_TRAVERSAL_IN)
1. ![MAJOR][MAJOR] [RskSystemProperties.java#L333](https://github.com/rsksmart/rskj/blob/d91111a0570e100175143f2b9b63c0669db7ce03/rskj-core/src/main/java/co/rsk/config/RskSystemProperties.java#L333): java/nio/file/Paths.get(Ljava/lang/String;[Ljava/lang/String;)Ljava/nio/file/Path; reads a file whose location might be specified by user input [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=findsecbugs%3APATH_TRAVERSAL_IN)
1. ![MAJOR][MAJOR] [RskSystemProperties.java#L335](https://github.com/rsksmart/rskj/blob/d91111a0570e100175143f2b9b63c0669db7ce03/rskj-core/src/main/java/co/rsk/config/RskSystemProperties.java#L335): java/nio/file/Paths.get(Ljava/lang/String;[Ljava/lang/String;)Ljava/nio/file/Path; reads a file whose location might be specified by user input [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=findsecbugs%3APATH_TRAVERSAL_IN)
1. ![MAJOR][MAJOR] [RskSystemProperties.java#L343](https://github.com/rsksmart/rskj/blob/d91111a0570e100175143f2b9b63c0669db7ce03/rskj-core/src/main/java/co/rsk/config/RskSystemProperties.java#L343): java/io/FileOutputStream.<init>(Ljava/lang/String;)V writes to a file whose location might be specified by user input [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=findsecbugs%3APATH_TRAVERSAL_OUT)
1. ![MAJOR][MAJOR] [DefaultConfig.java#L75](https://github.com/rsksmart/rskj/blob/d91111a0570e100175143f2b9b63c0669db7ce03/rskj-core/src/main/java/org/ethereum/config/DefaultConfig.java#L75): java/io/File.<init>(Ljava/lang/String;)V reads a file whose location might be specified by user input [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=findsecbugs%3APATH_TRAVERSAL_IN)
1. ![MAJOR][MAJOR] [SystemProperties.java#L87](https://github.com/rsksmart/rskj/blob/d91111a0570e100175143f2b9b63c0669db7ce03/rskj-core/src/main/java/org/ethereum/config/SystemProperties.java#L87): Method org.ethereum.config.SystemProperties.<static initializer for SystemProperties>() needlessly boxes a boolean constant [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3ANAB_NEEDLESS_BOOLEAN_CONSTANT_CONVERSION)
1. ![MAJOR][MAJOR] [SystemProperties.java#L119](https://github.com/rsksmart/rskj/blob/d91111a0570e100175143f2b9b63c0669db7ce03/rskj-core/src/main/java/org/ethereum/config/SystemProperties.java#L119): Incorrect lazy initialization and update of static field org.ethereum.config.SystemProperties.configFromFiles in new org.ethereum.config.SystemProperties(ConfigLoader) [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=findbugs%3ALI_LAZY_INIT_UPDATE_STATIC)
1. ![MAJOR][MAJOR] [SystemProperties.java#L121](https://github.com/rsksmart/rskj/blob/d91111a0570e100175143f2b9b63c0669db7ce03/rskj-core/src/main/java/org/ethereum/config/SystemProperties.java#L121): Method new org.ethereum.config.SystemProperties(ConfigLoader) passes a concatenated string to SLF4J's format string [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3ALO_APPENDED_STRING_IN_FORMAT_STRING)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by rskops** [comment]
SonarQube analysis reported 64 issues
* ![MAJOR][MAJOR] 37 major
* ![MINOR][MINOR] 8 minor
* ![INFO][INFO] 19 info

#### Top 10 issues

1. ![MAJOR][MAJOR] [RskSystemProperties.java#L298](https://github.com/rsksmart/rskj/blob/49ee8a01000aea604702d79821325bc4487021e1/rskj-core/src/main/java/co/rsk/config/RskSystemProperties.java#L298): Method co.rsk.config.RskSystemProperties.getRpcModules() does not presize the allocation of a collection [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3APSC_PRESIZE_COLLECTIONS)
1. ![MAJOR][MAJOR] [RskSystemProperties.java#L332](https://github.com/rsksmart/rskj/blob/49ee8a01000aea604702d79821325bc4487021e1/rskj-core/src/main/java/co/rsk/config/RskSystemProperties.java#L332): java/nio/file/Paths.get(Ljava/lang/String;[Ljava/lang/String;)Ljava/nio/file/Path; reads a file whose location might be specified by user input [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=findsecbugs%3APATH_TRAVERSAL_IN)
1. ![MAJOR][MAJOR] [RskSystemProperties.java#L333](https://github.com/rsksmart/rskj/blob/49ee8a01000aea604702d79821325bc4487021e1/rskj-core/src/main/java/co/rsk/config/RskSystemProperties.java#L333): java/nio/file/Paths.get(Ljava/lang/String;[Ljava/lang/String;)Ljava/nio/file/Path; reads a file whose location might be specified by user input [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=findsecbugs%3APATH_TRAVERSAL_IN)
1. ![MAJOR][MAJOR] [RskSystemProperties.java#L335](https://github.com/rsksmart/rskj/blob/49ee8a01000aea604702d79821325bc4487021e1/rskj-core/src/main/java/co/rsk/config/RskSystemProperties.java#L335): java/nio/file/Paths.get(Ljava/lang/String;[Ljava/lang/String;)Ljava/nio/file/Path; reads a file whose location might be specified by user input [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=findsecbugs%3APATH_TRAVERSAL_IN)
1. ![MAJOR][MAJOR] [RskSystemProperties.java#L343](https://github.com/rsksmart/rskj/blob/49ee8a01000aea604702d79821325bc4487021e1/rskj-core/src/main/java/co/rsk/config/RskSystemProperties.java#L343): java/io/FileOutputStream.<init>(Ljava/lang/String;)V writes to a file whose location might be specified by user input [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=findsecbugs%3APATH_TRAVERSAL_OUT)
1. ![MAJOR][MAJOR] [DefaultConfig.java#L75](https://github.com/rsksmart/rskj/blob/49ee8a01000aea604702d79821325bc4487021e1/rskj-core/src/main/java/org/ethereum/config/DefaultConfig.java#L75): java/io/File.<init>(Ljava/lang/String;)V reads a file whose location might be specified by user input [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=findsecbugs%3APATH_TRAVERSAL_IN)
1. ![MAJOR][MAJOR] [SystemProperties.java#L87](https://github.com/rsksmart/rskj/blob/49ee8a01000aea604702d79821325bc4487021e1/rskj-core/src/main/java/org/ethereum/config/SystemProperties.java#L87): Method org.ethereum.config.SystemProperties.<static initializer for SystemProperties>() needlessly boxes a boolean constant [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3ANAB_NEEDLESS_BOOLEAN_CONSTANT_CONVERSION)
1. ![MAJOR][MAJOR] [SystemProperties.java#L126](https://github.com/rsksmart/rskj/blob/49ee8a01000aea604702d79821325bc4487021e1/rskj-core/src/main/java/org/ethereum/config/SystemProperties.java#L126): new org.ethereum.config.SystemProperties(ConfigLoader) may fail to close stream [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=findbugs%3AOS_OPEN_STREAM)
1. ![MAJOR][MAJOR] [SystemProperties.java#L133](https://github.com/rsksmart/rskj/blob/49ee8a01000aea604702d79821325bc4487021e1/rskj-core/src/main/java/org/ethereum/config/SystemProperties.java#L133): Define and throw a dedicated exception instead of using a generic one. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS00112)
1. ![MAJOR][MAJOR] [SystemProperties.java#L176](https://github.com/rsksmart/rskj/blob/49ee8a01000aea604702d79821325bc4487021e1/rskj-core/src/main/java/org/ethereum/config/SystemProperties.java#L176): Define and throw a dedicated exception instead of using a generic one. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS00112)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by tinchou** [comment]
@joaquin-fdt do you mind waiting until the next PR?

It ends up being like this:

```java
private Config getNetworkDefaultConfig(Config userConfig) {
    if (userConfig.hasPath(SystemProperties.PROPERTY_BC_CONFIG_NAME)) {
        String network = userConfig.getString(SystemProperties.PROPERTY_BC_CONFIG_NAME);
        if (NodeCliFlags.NETWORK_TESTNET.getName().equals(network)) {
            return ConfigFactory.load(TESTNET_RESOURCE_PATH);
        } else if (NodeCliFlags.NETWORK_REGTEST.getName().equals(network)) {
            return ConfigFactory.load(REGTEST_RESOURCE_PATH);
        } else {
            logger.info("Invalid network '{}', using mainnet by default", network);
        }
    } else {
        logger.info("Network not set, using mainnet by default");
    }

    return ConfigFactory.load(MAINNET_RESOURCE_PATH);
}
```

File: rskj-core/src/main/java/org/ethereum/config/SystemProperties.java [hunk 3]
```diff
@@ -117,15 +113,14 @@
...
         try {
-            // could be locked but the result should be the same if there is no race condition
-            if (configFromFiles == null){
-                configFromFiles = getConfigFromFiles();
-                logger.debug("Config trace: " + configFromFiles.root().render(ConfigRenderOptions.defaults().
-                        setComments(false).setJson(false)));
-                validateConfig();
-            }
+            this.configFromFiles = loader.getConfigFromFiles();
+            logger.debug(
```
Comment (diega) [inline rskj-core/src/main/java/org/ethereum/config/SystemProperties.java]: What if we make it in _trace_?

File: rskj-core/src/main/java/co/rsk/config/ConfigLoader.java [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+
+        String file = System.getProperty("rsk.conf.file");
+        Config cmdLineConfigFile = file != null ? ConfigFactory.parseFile(new File(file)) : ConfigFactory.empty();
+        logger.info(
+                "Config ( {} ): user properties from -Drsk.conf.file file '{}'",
+                cmdLineConfigFile.entrySet().isEmpty() ? NO : YES,
+                file
+        );
+
+        Config userConfig = ConfigFactory.systemProperties()
```
Comment (diega) [inline rskj-core/src/main/java/co/rsk/config/ConfigLoader.java]: Shuldn't _systemProperties_ be the last fallback?

File: rskj-core/src/main/java/co/rsk/config/ConfigLoader.java [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+                .withFallback(cmdLineConfigFile)
+                .withFallback(installerConfig);
+        Config networkBaseConfig = getNetworkBaseConfig(userConfig);
+        return userConfig.withFallback(networkBaseConfig);
```
Comment (diega) [inline rskj-core/src/main/java/co/rsk/config/ConfigLoader.java]: I think we will be overriding userConfigs with networkBaseConfigs here, maybe it should go the other way around `networkBaseConfig.withFallback(userConfig)`

Comment (tinchou) [reply rskj-core/src/main/java/co/rsk/config/ConfigLoader.java]: Not really. Fallback is like the opposite of override.

File: rskj-core/src/main/java/co/rsk/config/ConfigLoader.java (hunk 4 above)
Comment (tinchou) [reply rskj-core/src/main/java/co/rsk/config/ConfigLoader.java]: fallback != override

File: rskj-core/src/main/java/co/rsk/config/ConfigLoader.java (hunk 5 above)
Comment (diega) [reply rskj-core/src/main/java/co/rsk/config/ConfigLoader.java]: gotcha

File: rskj-core/src/main/java/co/rsk/config/ConfigLoader.java [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+    private Config getNetworkBaseConfig(Config userConfig) {
+        // these read reference.conf automatically, and overlay the network config on top
+        if (userConfig.hasPath("blockchain.config.name")) {
+            String network = userConfig.getString("blockchain.config.name");
+            if ("testnet".equals(network)) {
+                return ConfigFactory.load("config/testnet");
+            } else if ("regtest".equals(network)) {
+                return ConfigFactory.load("config/regtest");
+            } else {
+                logger.info("Invalid network '{}', using mainnet by default", network);
```
[the remark this answers is not in the document]
Comment (tinchou) [reply rskj-core/src/main/java/co/rsk/config/ConfigLoader.java]: Agree

File: rskj-core/src/main/java/co/rsk/config/ConfigLoader.java [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+            }
+        } else {
+            logger.info("Network not set, using mainnet by default");
```
[the remark this answers is not in the document]
Comment (tinchou) [reply rskj-core/src/main/java/co/rsk/config/ConfigLoader.java]: I don't really agree here. It's supposed to be a default because is the most useful setting when a regular user is running a node.

[the remark this answers is not in the document]
Comment (diega) [reply rskj-core/src/main/java/co/rsk/config/ConfigLoader.java]: I get your point but I think this default is important enough to distinguish it from any other _info_ message, I understand the _warning_ name sounds misleading, but we are missing a method like _putRealAttentionToThis_ in log4j 😅. This is really subjective and doesn't look wrong as it is at all

**Comment by rskops** [comment]
SonarQube analysis reported 64 issues
* ![MAJOR][MAJOR] 37 major
* ![MINOR][MINOR] 8 minor
* ![INFO][INFO] 19 info

#### Top 10 issues

1. ![MAJOR][MAJOR] [RskSystemProperties.java#L298](https://github.com/rsksmart/rskj/blob/c12cb8045e97f2406b3ef6bb46e4d5fa0a6e26e0/rskj-core/src/main/java/co/rsk/config/RskSystemProperties.java#L298): Method co.rsk.config.RskSystemProperties.getRpcModules() does not presize the allocation of a collection [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3APSC_PRESIZE_COLLECTIONS)
1. ![MAJOR][MAJOR] [RskSystemProperties.java#L332](https://github.com/rsksmart/rskj/blob/c12cb8045e97f2406b3ef6bb46e4d5fa0a6e26e0/rskj-core/src/main/java/co/rsk/config/RskSystemProperties.java#L332): java/nio/file/Paths.get(Ljava/lang/String;[Ljava/lang/String;)Ljava/nio/file/Path; reads a file whose location might be specified by user input [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=findsecbugs%3APATH_TRAVERSAL_IN)
1. ![MAJOR][MAJOR] [RskSystemProperties.java#L333](https://github.com/rsksmart/rskj/blob/c12cb8045e97f2406b3ef6bb46e4d5fa0a6e26e0/rskj-core/src/main/java/co/rsk/config/RskSystemProperties.java#L333): java/nio/file/Paths.get(Ljava/lang/String;[Ljava/lang/String;)Ljava/nio/file/Path; reads a file whose location might be specified by user input [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=findsecbugs%3APATH_TRAVERSAL_IN)
1. ![MAJOR][MAJOR] [RskSystemProperties.java#L335](https://github.com/rsksmart/rskj/blob/c12cb8045e97f2406b3ef6bb46e4d5fa0a6e26e0/rskj-core/src/main/java/co/rsk/config/RskSystemProperties.java#L335): java/nio/file/Paths.get(Ljava/lang/String;[Ljava/lang/String;)Ljava/nio/file/Path; reads a file whose location might be specified by user input [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=findsecbugs%3APATH_TRAVERSAL_IN)
1. ![MAJOR][MAJOR] [RskSystemProperties.java#L343](https://github.com/rsksmart/rskj/blob/c12cb8045e97f2406b3ef6bb46e4d5fa0a6e26e0/rskj-core/src/main/java/co/rsk/config/RskSystemProperties.java#L343): java/io/FileOutputStream.<init>(Ljava/lang/String;)V writes to a file whose location might be specified by user input [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=findsecbugs%3APATH_TRAVERSAL_OUT)
1. ![MAJOR][MAJOR] [DefaultConfig.java#L75](https://github.com/rsksmart/rskj/blob/c12cb8045e97f2406b3ef6bb46e4d5fa0a6e26e0/rskj-core/src/main/java/org/ethereum/config/DefaultConfig.java#L75): java/io/File.<init>(Ljava/lang/String;)V reads a file whose location might be specified by user input [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=findsecbugs%3APATH_TRAVERSAL_IN)
1. ![MAJOR][MAJOR] [SystemProperties.java#L87](https://github.com/rsksmart/rskj/blob/c12cb8045e97f2406b3ef6bb46e4d5fa0a6e26e0/rskj-core/src/main/java/org/ethereum/config/SystemProperties.java#L87): Method org.ethereum.config.SystemProperties.<static initializer for SystemProperties>() needlessly boxes a boolean constant [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3ANAB_NEEDLESS_BOOLEAN_CONSTANT_CONVERSION)
1. ![MAJOR][MAJOR] [SystemProperties.java#L126](https://github.com/rsksmart/rskj/blob/c12cb8045e97f2406b3ef6bb46e4d5fa0a6e26e0/rskj-core/src/main/java/org/ethereum/config/SystemProperties.java#L126): new org.ethereum.config.SystemProperties(ConfigLoader) may fail to close stream [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=findbugs%3AOS_OPEN_STREAM)
1. ![MAJOR][MAJOR] [SystemProperties.java#L133](https://github.com/rsksmart/rskj/blob/c12cb8045e97f2406b3ef6bb46e4d5fa0a6e26e0/rskj-core/src/main/java/org/ethereum/config/SystemProperties.java#L133): Define and throw a dedicated exception instead of using a generic one. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS00112)
1. ![MAJOR][MAJOR] [SystemProperties.java#L176](https://github.com/rsksmart/rskj/blob/c12cb8045e97f2406b3ef6bb46e4d5fa0a6e26e0/rskj-core/src/main/java/org/ethereum/config/SystemProperties.java#L176): Define and throw a dedicated exception instead of using a generic one. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS00112)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'
